joblib/_parallel_backends.py

```diff
@@ -295,7 +295,7 @@ def __init__(self, batch):
         # arguments in memory
         self.results = batch()
 
-    def get(self):
+    def get(self, timeout=None):
```

Comment:
Maybe the documentation should be explicit that timeout only works when `n_jobs != 1`.